src/monodroid/jni/embedded-assemblies.cc

```diff
@@ -16,77 +16,59 @@ extern "C" {
 #include "monodroid.h"
 #include "dylib-mono.h"
 #include "util.h"
-#include "unzip.h"
-#include "ioapi.h"
 #include "embedded-assemblies.h"
 #include "globals.h"
 #include "monodroid-glue.h"
 
+namespace xamarin { namespace android { namespace internal {
+	struct TypeMappingInfo {
+		char                     *source_apk;
+		char                     *source_entry;
+		int                       num_entries;
+		int                       entry_length;
+		int                       value_offset;
+		const   char             *mapping;
+		struct  TypeMappingInfo  *next;
+	};
+
+	struct md_mmap_info {
+		void   *area;
+		size_t  size;
+	};
+}}}
+
 using namespace xamarin::android;
 using namespace xamarin::android::internal;
 
-struct TypeMappingInfo {
-	char                     *source_apk;
-	char                     *source_entry;
-	int                       num_entries;
-	int                       entry_length;
-	int                       value_offset;
-	const   char             *mapping;
-	struct  TypeMappingInfo  *next;
+const char *EmbeddedAssemblies::suffixes[] = {
+	"",
+	".dll",
+	".exe",
 };
 
-static  MonoBundledAssembly         **bundled_assemblies;
-static  int                           bundled_assemblies_count;
-static  monodroid_should_register     should_register;
-static  void                         *should_register_data;
-static  int                           register_debug_symbols;
-static  char                         *assemblies_prefix;
+constexpr char EmbeddedAssemblies::assemblies_prefix[];
 
-static  struct TypeMappingInfo       *java_to_managed_maps;
-static  struct TypeMappingInfo       *managed_to_java_maps;
-
-MONO_API void
-monodroid_embedded_assemblies_set_should_register (monodroid_should_register r, void *d)
+MonoAssembly*
+EmbeddedAssemblies::open_from_bundles (MonoAssemblyName* aname, bool ref_only)
 {
-	should_register       = r;
-	should_register_data  = d;
-}
-
-MONO_API int
-monodroid_embedded_assemblies_set_register_debug_symbols (int new_value)
-{
-	int cur                 = register_debug_symbols;
-	register_debug_symbols  = new_value;
-	return cur;
-}
-
-static MonoAssembly*
-open_from_bundles (MonoAssemblyName *aname, char **assemblies_path, void *user_data, mono_bool ref_only)
-{
-	DylibMono   *mono     = reinterpret_cast<DylibMono*> (user_data);
-	const char  *culture  = reinterpret_cast<const char*> (mono->assembly_name_get_culture (aname));
-	char *name;
-	char *ename;
-	size_t si;
-
-	MonoAssembly *a = nullptr;
+	const char *culture = reinterpret_cast<const char*> (monoFunctions.assembly_name_get_culture (aname));
+	const char *asmname = reinterpret_cast<const char*> (monoFunctions.assembly_name_get_name (aname));
```

Comment:
Ditto here:

```cpp
const char*
DylibMono::assembly_name_get_name (MonoAssemblyName *aname)
```

`DylibMono::assembly_name_get_name()` *already* returns a `const char*`.  Why is there a cast?